hydra/functions/FlatteLineShape.h

```diff
@@ -78,11 +79,13 @@ namespace hydra {
  * @param daugther3_mass daughter particle 2 mass
  * @param radi decay vertex radio.
  */
-        FlatteLineShape(hydra::Parameter const& mean, hydra::Parameter const& rho1 , hydra::Parameter const& rho2,
```

Comment:
What is the length of the vector `std::vector<std::vector<double>> params` and what does it holds?
I think can be a good idea to document the meaning of this parameter. 
Also, are these parameters supposed to be optimized in a fit? Case yes, consider to use the correct interface to store and make them "trackable" by Minuit2 and copy constructors on device and host sides. 
If the number of such parameters is not fixed, but still constant at compile time, please, add a template parameter to the class to implement then, and set the default value to the most common case (my guess is most of the time it is 2...) . After set them using `this->SetParameter(param_index, param_value) ` in the body of the constructor.